build/build.cake

```diff
@@ -98,24 +100,28 @@ static bool IsExcludedDirectory(FilePath path)
     return
         segments.Contains("bin") ||
         segments.Contains("internal") ||
-        segments.Contains("obj");
+        segments.Contains("obj") ||
+        segments.Contains("Generated Files");
 }
 
 // Returns true if the given file is source that the build system
 // should use directly i.e. it is not generated in the build and
 // is not being excluded for some reason.
 static bool IsBuildInput(FilePath path)
 {
-    return !IsExcludedDirectory(path) && !IsAutoGenerated(path);
+    var filename = path.GetFilename().ToString();
```

Comment:
>ToString [](start = 38, length = 8)

Don't convert to a string. Keep the strong type and use the FilePath.GetExtension() method to test the extension.